torch/optim/adamax.py

```diff
@@ -53,13 +56,12 @@ def __setstate__(self, state):
             group.setdefault("maximize", False)
             group.setdefault("differentiable", False)
             group.setdefault("capturable", False)
-        state_values = list(self.state.values())
-        step_is_tensor = (len(state_values) != 0) and torch.is_tensor(
-            state_values[0]["step"]
-        )
-        if not step_is_tensor:
-            for s in state_values:
-                s["step"] = torch.tensor(float(s["step"]), dtype=_get_scalar_dtype())
+            for p in group["params"]:
+                p_state = self.state.get(p, [])
+                if len(p_state) != 0 and not torch.is_tensor(p_state['step']):
```

Comment:
```suggestion
                if not torch.is_tensor(p_state.get('step', None)):
```

Reply:
This is not equivalent! If p_state is empty, we don't want to modify any state, but the conditional will run in your suggestion, right?

Reply:
torch.is_tensor(None) should still be False and so won't go in?

Reply:
Ho right, my bad!